include/boost/geometry/algorithms/detail/envelope/linestring.hpp

```diff
@@ -144,19 +154,27 @@ struct envelope_linear_on_spheroid
                 typename coordinate_system<Box>::type::units
             >::apply(longitude_intervals, lon_min, lon_max);
 
-        assign_values(mbr, lon_min, lat_min, lon_max, lat_max);
+        geometry::set<min_corner, 0>(mbr, lon_min);
+        geometry::set<min_corner, 1>(mbr, lat_min);
+        geometry::set<max_corner, 0>(mbr, lon_max);
+        geometry::set<max_corner, 1>(mbr, lat_max);
+
+        dispatch::envelope
+            <
+                Linear, 2, DimensionCount
+            >::apply(linear, mbr);
```

Comment:
Ok I see the purpose of having the Dimensions here. First the segments must be handled for the first 2 coordinates and then for the rest of the coordinates points can be used as before. Still I think that it shouldn't be done on the `envelope` level. Mainly because of the "obscured" handling of coordinates if Dimension is 0.

Btw, AFAIU this algorithm is called for bot Linestrings and MultiLinestrings. And for both of those geometries the Segments are gathered in a separate container (which is twice as big as the initial Geometry!). Correct me if I'm wrong but must we search for gaps using Segments? I think that with Single geometry it's clear what is the Box because it's closed, and the segments between two arbitrary points on a sphere are well defined (the shortest). So for a Single geometry you just need to traverse the segments and expand the Box. Only for MultiGeometry the gaps must be found, and the intervals are then the envelopes of SingleGeometries, not segments. This reduces the complexity significantly.